diff --git a/zipFile.js b/zipFile.js
index 234f6cdbfa954c077c31000459e0d7c22c1533f4..39e23f1a9662b2ccff6eef96761bca0f93c9998e 100644
--- a/zipFile.js
+++ b/zipFile.js
@@ -294,7 +294,10 @@ module.exports = function (/*Buffer|null*/ inBuffer, /** object */ options) {
 
             mainHeader.size = 0;
             mainHeader.offset = 0;
-            totalEntries = 0;
+
+            // A fix of https://github.com/cthackers/adm-zip/issues/526.
+            // TODO: Drop the patch once the issue is resolved.
+            let totalEntries = 0;
 
             for (const entry of this.entries) {
                 // compress data and set local and entry header accordingly. Reason why is called first
